tests/test-class-wc-payments-customer-service.php

```diff
@@ -100,6 +114,28 @@ public function test_create_customer_for_user() {
 		$customer_id = $this->customer_service->create_customer_for_user( $user, 'Test User', 'test.user@example.com' );
 
 		$this->assertEquals( 'cus_test12345', $customer_id );
+		$this->assertEquals( 'cus_test12345', get_user_option( '_wcpay_customer_id', $user->ID ) );
+	}
+
+	/**
+	 * Test create customer for user for test mode.
+	 *
+	 * @throws WC_Payments_API_Exception
+	 */
+	public function test_create_customer_for_user_test_mode() {
+		WC_Payments::get_gateway()->update_option( 'test_mode', 'yes' );
+		$user             = new WP_User( 1 );
+		$user->user_login = 'testUser';
+
+		$this->mock_api_client->expects( $this->once() )
+			->method( 'create_customer' )
+			->with( 'Test User', 'test.user@example.com', 'Name: Test User, Username: testUser' )
+			->willReturn( 'cus_test12345' );
+
+		$customer_id = $this->customer_service->create_customer_for_user( $user, 'Test User', 'test.user@example.com' );
+
+		$this->assertEquals( 'cus_test12345', $customer_id );
+		$this->assertEquals( 'cus_test12345', get_user_option( '_wcpay_customer_id_test', $user->ID ) );
```

Comment:
It would be nice to make sure the `live` id was not set for this user.

Reply:
Good idea! Added the check for both tests in https://github.com/Automattic/woocommerce-payments/pull/918/commits/6cfa1142b6460b901a0f8d52a98f31b254cc6e89.